src/Plugin/resource/CachedResource.php

```diff
@@ -191,14 +206,24 @@ public function getControllers() {
    * {@inheritdoc}
    */
   public function index($path) {
-    return $this->subject->index($path);
+    return $this->getDataProvider()->index();
```

Comment:
> I think that it would make it more readable to have a $this->getPrimaryResource()->index()

:+1: I'm cool with that. Anyways `subject` is the nomenclature for the decorator pattern. I will change this to `$this->primaryResource->index()` since I'd rather keep the access to the decorated resource protected.

> I think we can remove all the other proxy methods

We need to keep the proxy methods, since this class implements `ResourceInterface` and should be used as any other resource. What we could do is create a `DecoratedResourceBase` abstract class that has all of the proxy methods. All resource decorators would extend from that one.